### PR TITLE
uds only starts once and then gets kicked

### PR DESCRIPTION
**Pull request by ilackarms**
fix race when reconcile called more than once
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/631

### CODE REVIEW

File: projects/discovery/pkg/uds/syncer/discovery_syncer.go [hunk 1]
```diff
@@ -34,26 +31,6 @@ func (s *syncer) Sync(ctx context.Context, snap *v1.DiscoverySnapshot) error {
...
-				contextutils.LoggerFrom(ctx).Errorf("error in UDS: %v", err)
-			case <-ctx.Done():
-				return
-			}
-		}
-	}()
-
-	return nil
+	// kick the uds, ensure that desired upstreams are in sync
+	return s.uds.Resync(ctx)
```
Comment (yuval-k) [inline projects/discovery/pkg/uds/syncer/discovery_syncer.go]: i'm not sure i understand why resync is called here.
it seems - that the state that rsync depends on gets updated by the upstream discovery, but StartUds already calls resync when that state changed  - so it looks redundant here?

File: test/kube2e/gateway/gateway_test.go [hunk 2]
```diff
@@ -62,18 +69,31 @@ var _ = Describe("Kube2e: gateway", func() {
...
+		upstreamClient, err = gloov1.NewUpstreamClient(upstreamClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		err = upstreamClient.Register()
+		Expect(err).NotTo(HaveOccurred())
 	})
 
 	AfterEach(func() {
 		cancel()
-		err := virtualServiceClient.Delete(testHelper.InstallNamespace, "vs", clients.DeleteOpts{})
-		Expect(err).NotTo(HaveOccurred())
```
Comment (rickducott) [inline test/kube2e/gateway/gateway_test.go]: Why did you remove this expect?

Comment (yuval-k) [reply test/kube2e/gateway/gateway_test.go]: because the stress tests dont have a "vs" so this delete fails.
